timely/src/dataflow/channels/pushers/tee.rs

```diff
@@ -4,26 +4,29 @@ use std::cell::RefCell;
 use std::fmt::{self, Debug};
 use std::rc::Rc;
 
-use crate::dataflow::channels::{Bundle, Message};
-use crate::Data;
+use crate::dataflow::channels::{BundleCore, Message};
 
 use crate::communication::Push;
+use crate::{Container, Data};
 
-type PushList<T, D> = Rc<RefCell<Vec<Box<dyn Push<Bundle<T, D>>>>>>;
+type PushList<T, D> = Rc<RefCell<Vec<Box<dyn Push<BundleCore<T, D>>>>>>;
 
 /// Wraps a shared list of `Box<Push>` to forward pushes to. Owned by `Stream`.
-pub struct Tee<T: 'static, D: 'static> {
-    buffer: Vec<D>,
+pub struct TeeCore<T, D> {
+    buffer: D,
     shared: PushList<T, D>,
 }
 
-impl<T: Data, D: Data> Push<Bundle<T, D>> for Tee<T, D> {
+/// [TeeCore] specialized to `Vec`-based container.
+pub type Tee<T, D> = TeeCore<T, Vec<D>>;
+
+impl<T: Data, D: Container> Push<BundleCore<T, D>> for TeeCore<T, D> {
     #[inline]
-    fn push(&mut self, message: &mut Option<Bundle<T, D>>) {
+    fn push(&mut self, message: &mut Option<BundleCore<T, D>>) {
         let mut pushers = self.shared.borrow_mut();
         if let Some(message) = message {
             for index in 1..pushers.len() {
-                self.buffer.extend_from_slice(&message.data);
+                self.buffer.clone_from(&message.data);
```

Comment:
Change of behavior: In the past, all buffers had at least default capacity. After this change, the buffers passed around might be oddly sized.